test/services/helpers/service.test.js

```
@@ -1,74 +1,52 @@
+const MailService = require('../../../src/services/helpers/service');
 
-	before((done) => {
-		// Enable mockery to mock objects
-		mockery.enable({
-			warnOnUnregistered: false,
-		});
+// eslint-disable-next-line import/no-dynamic-require
+const config = require(`../../../config/${process.env.NODE_ENV || 'default'}.json`);
 
-		// Once mocked, any code that calls require('nodemailer') will get our nodemailerMock
-		mockery.registerMock('nodemailer', nodemailerMock);
+const isMailbodyValid = ({
+	platform,
+	to,
+	subject,
+	text,
+	html,
+	replyTo,
+}) => !!(platform && to && subject && (text || html) && replyTo);
 
-		// Make sure anything that uses nodemailer is loaded here, after it is mocked...
-		delete require.cache[require.resolve('../../../src/services/helpers/service')];
+describe('Mail Service', () => {
+	let mailService;
 
-		// load new app to make mockery work
-		const feathers = require('@feathersjs/feathers');
-		const app = feathers();
-		let secrets;
-		try {
-			secrets = require('../config/secrets.json');
-		} catch (error) {
-			secrets = {};
-		}
-		app.set('secrets', secrets);
-		const MailService = require('../../../src/services/helpers/service')(app);
+	before(() => {
 		app.use('/mails', new MailService());
 		mailService = app.service('/mails');
-
-		done();
 	});
 
-	after(() => {
-		// Remove our mocked nodemailer and disable mockery
-		mockery.deregisterAll();
-		mockery.disable();
-	});
-
-	it('should send an email using nodemailer-mock', () => {
-		mailService.create(emailOptions)
-			.then((_) => {
-				// get the array of emails we sent
-				const sentMail = nodemailerMock.mock.sentMail();
-				expect(sentMail.length).to.equal(1);
-				expect(sentMail[0].to).to.equal(emailOptions.email);
-				expect(sentMail[0].subject).to.equal(emailOptions.subject);
-				expect(sentMail[0].html).to.equal(emailOptions.content.html);
-				expect(sentMail[0].text).to.equal(emailOptions.content.text);
+	describe('valid emails', () => {
```

maybe add one negative test? email without recipient or something?

---

and in this case, the mail service should throw an error I'd reckon